### PR TITLE
feat: adds hidden field functionality to surveys

### PR DESCRIPTION
**Pull request by rtpa25**
## What does this PR do?

Fixes (https://github.com/formbricks/formbricks/issues/887)

https://www.loom.com/share/5b5ecc20a80241c3aedb1c3f3d8bac69

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change adds a new database migration
- [ ] This change requires a documentation update

### Required

- [x] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [x] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated

- [x] If a UI change was made: Added a screen recording or screenshots to this PR

### CODE REVIEW

**Comment by jobenjada** [comment]
Tweaked the UI but struggle to push it:

image

image

Also, if someone copies a value incl. a space, it can be submitted:
image


Looking forward to the final PR :)

**Comment by rtpa25** [comment]
Will get it done by tomorrow

File: apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/SurveyEditor.tsx [hunk 1]
```diff
@@ -38,6 +38,8 @@ export default function SurveyEditor({
 
   useEffect(() => {
     if (survey) {
+      console.log("survey", survey);
+
```
Comment (rtpa25) [inline apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/SurveyEditor.tsx:42]: clean

File: packages/database/.env [hunk 2]
```diff
@@ -0,0 +1 @@
+DATABASE_URL='postgresql://postgres:postgres@localhost:5432/formbricks?schema=public'
```
Comment (rtpa25) [inline packages/database/.env:1]: clean

**Review by rtpa25 — COMMENTED** [review]
LGTM

**Comment by rtpa25** [comment]
@jobenjada @mattinannt awaiting review

**Comment by mattinannt** [comment]
@rtpa25 Thanks a lot for the feature :-) 💪 Looks great!
We now have mix a different terms in Code and UI. Can you please rename the code variables and files that it's named `Hidden Field` instead of `Hidden Question` to be consistent?

**Comment by rtpa25** [comment]
Ya will do that shortly I did that deliberately to understand which side
are u more inclined to field or question

**Comment by rtpa25** [comment]
Going with field

**Comment by rtpa25** [comment]
Done ☑️

**Review by mattinannt — APPROVED** [review]

**Pull request merged**